### PR TITLE
Extract a NonNativeFeeWarning component

### PR DESCRIPTION
**Pull request by jessepinho**
Per [this comment](https://github.com/penumbra-zone/web/pull/1268#discussion_r1643130786), this PR reduces duplication by extracting a `<NonNativeFeeWarning />` component which determines for itself whether to render, rather than having the parent component determine whether to render it.

### CODE REVIEW

File: apps/minifront/src/components/shared/non-native-fee-warning.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import { ReactNode } from 'react';
+
+const hasStakingToken = (
+  balancesResponses: BalancesResponse[] = [],
+  stakingAssetMetadata?: Metadata,
+): boolean => {
+  return balancesResponses.some(asset =>
+    getAssetIdFromValueView
+      .optional()(asset.balanceView)
+      ?.equals(getAssetId.optional()(stakingAssetMetadata)),
```
Comment (jessepinho) [inline apps/minifront/src/components/shared/non-native-fee-warning.tsx:15]: Weirdly, this is how Prettier formatted these lines — with a `?` at the beginning

File: apps/minifront/src/components/shared/non-native-fee-warning.tsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+export const useShouldRender = (balancesResponses: BalancesResponse[] = [], amount: number) => {
+  const stakingTokenMetadata = useStakingTokenMetadata();
+  const userHasStakingToken = hasStakingToken(balancesResponses, stakingTokenMetadata.data);
+  const showNonNativeFeeWarning = amount > 0 && !userHasStakingToken;
+
+  return showNonNativeFeeWarning;
+};
+
+/**
+ * Renders a non-native fee warning if A) the user has balance of the staking
```
Comment (TalDerei) [inline apps/minifront/src/components/shared/non-native-fee-warning.tsx:28]: A) "The user _doesn't_ have the native staking token"

**Review by TalDerei — APPROVED** [review]
modularity!

File: apps/minifront/src/components/shared/non-native-fee-warning.tsx (hunk 2 above)
Comment (jessepinho) [reply apps/minifront/src/components/shared/non-native-fee-warning.tsx]: Whoops! Good catch. Fixed.

**Pull request merged**